csrc/id_model/loop_promotion.cpp

```diff
@@ -1293,4 +1312,45 @@ std::unordered_map<ValGroup, IterDomain*> LoopPromotionMapBuilder::
   return map;
 }
 
+void LoopPromotionMapBuilder::revertBroadcastOnlyLoopGroups(
+    std::unordered_map<ValGroup, IterDomain*>& loop_promotion_map) const {
+  // If a loop group originally only consisted of broadcast IDs
+  // and now is promoted to a non-broadcast ID, it should not need to
+  // be promoted.
+  for (auto& [loop_group, current_promotion_id] : loop_promotion_map) {
+    if (current_promotion_id->isBroadcast()) {
+      continue;
+    }
+
+    // As long as there's a single ID marked as broadcat only, this
+    // group originally consisted of broadcast IDs only. Note that,
+    // since new IDs were added as part of the promotion analysis, not
+    // all of the IDs may not be included in the broadcast only set.
```

Comment:
nit: Do you mean the following?
```suggestion
    // all of the IDs are included in the broadcast only set.
```